src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writestatementofreasons/WriteStatementOfReasonsAboutToSubmitHandler.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.writestatementofreasons;
+
+import static java.util.Objects.requireNonNull;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.DocumentType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.sscs.service.FooterService;
+import uk.gov.hmcts.reform.sscs.util.SscsUtil;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class WriteStatementOfReasonsAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+    @Value("${feature.postHearings.enabled}")
+    private final boolean isPostHearingsEnabled;
+    private final FooterService footerService;
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbackType must not be null");
+
+        return callbackType.equals(CallbackType.ABOUT_TO_SUBMIT)
+            && callback.getEvent() == EventType.SOR_WRITE
+            && isPostHearingsEnabled;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
+        Callback<SscsCaseData> callback,
+        String userAuthorisation
+    ) {
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+        String caseId = caseData.getCcdCaseId();
+        log.info("Write Statement of Reasons: handling action for case {}", caseId);
+
+        final PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
+
+        if (response.getErrors().isEmpty()) {
```

Comment:
missing bit of CC on this line

Reply:
I have not been able to find a way to trigger an error in the test. Given that this handler structure is the same as PostHearingReviewAboutToSubmitHandler, and the test class for that file also does not cover this condition, I am going to accept this as it is.